### PR TITLE
Python code style guide

### PR DESCRIPTION
**Pull request by frinzell**
Signed-off-by: Frinzell, Aaron <aaron.frinzell@intel.com>

### CODE REVIEW

File: docs/sphinx/contribution/code-style-python.rst [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+
+#. Headers and Comments
+
+    Use single line comments, a single hash/number sign/octothorpe '#'.
+
+    Should contain a space immediately after the '#'.
+
+    .. code-block:: python
+        # Good header comment
```
Comment (Sae86) [inline docs/sphinx/contribution/code-style-python.rst:39]: May need a space between code block directive and comment

File: docs/sphinx/contribution/code-style-python.rst [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+
+#. Single vs Double Quotes
+
+    Single quotes are encouraged but can vary with use case.
+
+    Avoid using backslashes '\\' in strings.
+
+    .. code-block:: python
+        'This is a preferred "string".'
```
Comment (Sae86) [inline docs/sphinx/contribution/code-style-python.rst:49]: Same here

File: docs/sphinx/contribution/code-style-python.rst [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+    Import order:
+        #. Python standard library
+        #. Third-party imports
+        #. CHIPSEC and local application imports
+
+    Avoid using ``import *`` or ``from import *``.  This could pollute the namespace.
+
+    .. code-block:: python
+        # Good
```
Comment (Sae86) [inline docs/sphinx/contribution/code-style-python.rst:65]: And here

File: docs/sphinx/contribution/code-style-python.rst [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+
+#. Operator Precedence, Comparisons, and Parentheses
+
+    If in doubt, wrap evaluated operators into logical sections if using multiple operators or improves readability.
+
+    While not needed in most cases, it can improve readability and limit the possibility of 'left-to-right chaining' issues.
+
+    .. code-block:: python
+        # Preferred
```
Comment (Sae86) [inline docs/sphinx/contribution/code-style-python.rst:139]: Also here

File: docs/sphinx/contribution/code-style-python.rst [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+    No trailing whitespace.
+
+
+#. Non-ASCII Characters
+
+    If including any non-ASCII characters anywhere in a python file, include the python encoding comment at the beginning of the file.
+
+    .. code-block:: python
+        # -*- coding: utf-8 -*-
```
Comment (Sae86) [inline docs/sphinx/contribution/code-style-python.rst:166]: Here as well

File: docs/sphinx/contribution/code-style-python.rst [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+
+    No non-ASCII class, function, or variable names.
+
+
+#. Docstrings
+
+    Use three double-quotes for module scoped docstrings.
+
+    .. code-block:: python
+        """String description of module or docstring."""
```
Comment (Sae86) [inline docs/sphinx/contribution/code-style-python.rst:176]: and here

File: docs/sphinx/contribution/code-style-python.rst (hunk 1 above)
Comment (frinzell) [reply docs/sphinx/contribution/code-style-python.rst:40]: Thanks! They should all be resolved.

File: docs/sphinx/contribution/code-style-python.rst [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+    Avoid using ``from __future__ imports``.  These may not work on older or all interpreter versions required in all supported environments.
+
+
+#. Line Length
+
+    Maximum line length should be 120 characters.
+
+    If at or near this limit, consider rewriting (eg simplifying) the line instead of breaking it into multiple lines.
+
+    Long lines can be an indication that too many things are happening at once and/or to difficult to read.
```
Comment (npmitche) [inline docs/sphinx/contribution/code-style-python.rst:86]:
```suggestion
    Long lines can be an indication that too many things are happening at once and/or too difficult to read.
```
After writing the suggestion, I'm double guessing myself. Not sure which is right.

File: docs/sphinx/contribution/code-style-python.rst [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+
+#. Docstrings
+
+    Use three double-quotes for module scoped docstrings.
+
+    .. code-block:: python
+
+        """String description of module or docstring."""
+
+    Single quote for function or routine docstring.
```
Comment (npmitche) [inline docs/sphinx/contribution/code-style-python.rst:184]: Does this match with what we are thinking about wrt adding docstrings to the API functions? Or do those also need to be triple quote comments?

Comment (frinzell) [reply docs/sphinx/contribution/code-style-python.rst:184]: Mixed but mostly double but should clarify if keeping it.

**Review by Sae86 — APPROVED** [review]

**Pull request merged**